src/core/api/providers/bedrock.ts

```diff
@@ -266,25 +268,25 @@ export class AwsBedrockHandler implements ApiHandler {
 	}
 
 	/**
-	 * Gets the appropriate model ID, accounting for cross-region inference if enabled.
-	 * For custom models, returns the raw model ID without any encoding.
+	 * Gets the appropriate model ID using inference profiles.
+	 * Supports none, regional, and global inference strategies.
+	 * For custom models, returns the raw model ID without any inference profile.
 	 */
 	async getModelId(): Promise<string> {
-		if (!this.options.awsBedrockCustomSelected && this.options.awsUseCrossRegionInference) {
-			const regionPrefix = this.getRegion().slice(0, 3)
-			switch (regionPrefix) {
-				case "us-":
-					return `us.${this.getModel().id}`
-				case "eu-":
-					return `eu.${this.getModel().id}`
-				case "ap-":
-					return `apac.${this.getModel().id}`
-				default:
-					// cross region inference is not supported in this region, falling back to default model
-					return this.getModel().id
-			}
-		}
-		return this.getModel().id
+		const baseModelId = this.getModel().id
+		const region = this.getRegion()
+		const customSelected = this.options.awsBedrockCustomSelected || false
+
+		// Get inference profile mode (with backwards compatibility)
+		const inferenceProfileMode = InferenceProfileResolver.getInferenceProfileMode(
+			this.options.awsInferenceStrategy,
+			this.options.awsUseCrossRegionInference,
+		)
+
+		// Resolve the model ID using the inference profile system
+		const resolution = InferenceProfileResolver.resolveModelId(baseModelId, region, inferenceProfileMode, customSelected)
+
```

Comment:
In this PR I face an issue Specifically for none inference profile 

<img width="511" height="733" alt="Image" src="https://github.com/user-attachments/assets/1b7eb7c7-266e-4aa3-b667-f307915b7d5b" />


I got 

<img width="548" height="608" alt="Image" src="https://github.com/user-attachments/assets/5a5e2533-b833-4862-900d-ae28a8adfd48" />



After putting some debug logs here I found this in your PR.  


<img width="1201" height="359" alt="Image" src="https://github.com/user-attachments/assets/933abaf0-0523-4c9d-8ff2-c634013175e4" />


But this works fine before your PR

<img width="384" height="896" alt="Image" src="https://github.com/user-attachments/assets/ddc71694-5b86-4be7-a340-c8a975008c89" />


I assume that is because sonnet 4 model needs Global inference region correct?